rdrf/rdrf/forms/widgets/settings_widgets.py

```diff
@@ -1,242 +1,151 @@
 import json
 
 from django.forms import Widget
+from django.forms.renderers import get_default_renderer
 from django.utils.safestring import mark_safe
 from django.utils.translation import gettext as _
 
 from .widgets import TimeWidget
 
 
-class SliderWidgetSettings(Widget):
+class JSONWidget(Widget):
 
-    @staticmethod
-    def get_allowed_fields():
-        return {'min', 'max', 'left_label', 'right_label', 'step'}
-
-    @staticmethod
-    def generate_input(name, title, parsed, info=None):
-        value = parsed.get(name, '')
-        input_str = f'<input type="text" name="{name}" id="{name}" value="{value}" onchange="saveJSON()">'
-        help_text = f'<div class="help">{info}</div>' if info else ''
-        return f"""
-            <div>
-                <label for="{name}">{title}</label>
-                {input_str}
-                {help_text}
-            </div>"""
-
-    def generate_inputs(self, parsed):
-        rows = [
-            self.generate_input('min', _('Min value'), parsed, _("leave empty if you want to use the CDE's min value")),
-            self.generate_input('max', _('Max value'), parsed, _("leave empty if you want to use the CDE's max value")),
-            self.generate_input('left_label', _('Left label'), parsed),
-            self.generate_input('right_label', _('Right label'), parsed),
-            self.generate_input('step', _('Step'), parsed),
-        ]
-        return "<br/>".join(rows)
+    def get_allowed_fields(self):
+        pass
```

Comment:
As this method has to be overridden by all subclasses can we throw `NotImplementedError` instead of the `pass`?

Reply:
Updated.